Support for the RADclock and cumulative virtual timecounter
Freebsd Kernel patch: FreeBSD-CURRENT
Julien Ridoux
julien@synclab.org

diff -aur --unidirectional-new-file sys/amd64/amd64/tsc.c FreeBSD-CURRENT/amd64/amd64/tsc.c
--- sys/amd64/amd64/tsc.c	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/amd64/amd64/tsc.c	2010-12-09 16:27:51.000000000 +1100
@@ -2,6 +2,12 @@
  * Copyright (C) 1998-2003 Poul-Henning Kamp
  * All rights reserved.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/i386/i386/tsc.c FreeBSD-CURRENT/i386/i386/tsc.c
--- sys/i386/i386/tsc.c	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/i386/i386/tsc.c	2010-12-09 16:27:51.000000000 +1100
@@ -2,6 +2,12 @@
  * Copyright (C) 1998-2003 Poul-Henning Kamp
  * All rights reserved.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/i386/xen/clock.c FreeBSD-CURRENT/i386/xen/clock.c
--- sys/i386/xen/clock.c	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/i386/xen/clock.c	2010-12-09 16:27:51.000000000 +1100
@@ -5,6 +5,12 @@
  * This code is derived from software contributed to Berkeley by
  * William Jolitz and Don Ahn.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/kern/kern_tc.c FreeBSD-CURRENT/kern/kern_tc.c
--- sys/kern/kern_tc.c	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/kern/kern_tc.c	2010-12-09 16:27:51.000000000 +1100
@@ -5,8 +5,15 @@
  * can do whatever you want with this stuff. If we meet some day, and you think
  * this stuff is worth it, you can buy me a beer in return.   Poul-Henning Kamp
  * ----------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
  */
 
+
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD: src/sys/kern/kern_tc.c,v 1.191 2010/09/21 08:02:02 mav Exp $");
 
diff -aur --unidirectional-new-file sys/net/bpf.c FreeBSD-CURRENT/net/bpf.c
--- sys/net/bpf.c	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/net/bpf.c	2010-12-09 16:27:51.000000000 +1100
@@ -7,6 +7,12 @@
  * to Berkeley by Steven McCanne and Van Jacobson both of Lawrence
  * Berkeley Laboratory.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/net/bpf.h FreeBSD-CURRENT/net/bpf.h
--- sys/net/bpf.h	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/net/bpf.h	2010-12-09 16:27:51.000000000 +1100
@@ -7,6 +7,12 @@
  * to Berkeley by Steven McCanne and Van Jacobson both of Lawrence
  * Berkeley Laboratory.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/sys/time.h FreeBSD-CURRENT/sys/time.h
--- sys/sys/time.h	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/sys/time.h	2010-12-09 16:27:51.000000000 +1100
@@ -2,6 +2,12 @@
  * Copyright (C) 1982, 1986, 1993
  *	The Regents of the University of California.  All rights reserved.
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
  * are met:
diff -aur --unidirectional-new-file sys/sys/timepps.h FreeBSD-CURRENT/sys/timepps.h
--- sys/sys/timepps.h	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/sys/timepps.h	2010-12-09 16:27:51.000000000 +1100
@@ -6,6 +6,12 @@
  * this stuff is worth it, you can buy me a beer in return.   Poul-Henning Kamp
  * ----------------------------------------------------------------------------
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * $FreeBSD: src/sys/sys/timepps.h,v 1.19 2005/01/07 02:29:24 imp Exp $
  *
  * The is a FreeBSD version of the RFC 2783 API for Pulse Per Second 
diff -aur --unidirectional-new-file sys/sys/timetc.h FreeBSD-CURRENT/sys/timetc.h
--- sys/sys/timetc.h	2010-12-09 15:46:09.000000000 +1100
+++ FreeBSD-CURRENT/sys/timetc.h	2010-12-09 16:27:51.000000000 +1100
@@ -6,6 +6,12 @@
  * this stuff is worth it, you can buy me a beer in return.   Poul-Henning Kamp
  * ----------------------------------------------------------------------------
  *
+ * Copyright (C) 2010 The University of Melbourne
+ * All rights reserved.
+ *
+ * Portions of this software were developed by the University of Melbourne
+ * under sponsorship from the FreeBSD Foundation.
+ *
  * $FreeBSD: src/sys/sys/timetc.h,v 1.62 2010/09/14 08:48:06 mav Exp $
  */
 
